code/src/components/product-card/ProductCard.test.tsx

```diff
@@ -19,78 +20,77 @@ const mockProduct: Product = {
   price: 500
 };
 
-const mockCartIconClickHandler = jest.fn();
-const mockSetState = jest.fn();
+const mockAddToCartOrOpenDrawer = jest.fn();
 
-jest.mock("react", () => ({
-  ...jest.requireActual("react"),
-  useState: jest.fn((init) => [init, mockSetState])
-}));
+jest.mock("@/hooks/use-add-to-cart-or-open-drawer/useAddToCartOrOpenDrawer");
 
-describe("ProductCard component", () => {
-  beforeEach(() => {
-    renderWithProviders(
-      <ProductCard
-        product={mockProduct}
-        onCartIconClick={mockCartIconClickHandler}
-        isUserAuthorized
-        isInCart
-      />
-    );
+const renderAndMock = (isProductInCart: boolean) => {
+  (useAddToCartOrOpenDrawer as jest.Mock).mockReturnValue({
+    isProductInCart,
+    addToCartOrOpenDrawer: mockAddToCartOrOpenDrawer
   });
+  renderWithProviders(<ProductCard product={mockProduct} />);
+};
 
+describe("ProductCard component", () => {
```

Comment:
naming. you noticed that name `component` should't be as a name of tests block. remove it

Reply:
That describe name was not writted by me, but thank you for noticing, I fixed it!